cub/warp/warp_exchange.cuh

```diff
@@ -468,5 +468,89 @@ public:
   //@}  end member group
 };
 
+template <typename InputT, int ITEMS_PER_THREAD>
+class WarpExchangeRegister
+{
+  static_assert(PowerOfTwo<ITEMS_PER_THREAD>::VALUE, "ITEMS_PER_THREAD must be a power of two");
+
+  template <int SIZE, int idx>
+  class CompileTimeArrayImpl; // fwd decl for partial specializations
+
+  template <int SIZE> // terminating partial specialization
```

Comment:
If `SIZE` is always equal to `ITEMS_PER_THREAD`, we can use `ITEMS_PER_THREAD` directly and remove extra template parameter. This might remove some mental load while reading the code. To workaround partial specialization issue, you could introduce a `DUMMY` parameter, like [here](https://github.com/NVIDIA/cub/blob/6fa6d6e7715d43da8605c7b6766ca51035bdf4ec/cub/block/block_load.cuh#L663).

Reply:
I actually tried that, the problem is `nvcc` seems to disallow explicit specialization of template classes in class scope. So I guess the only solution here is to use `DUMMY`?

Reply:
This is rather C++ problem, explicit specializations are not allowed at class scope. As I suggested above, feel free to use `DUMMY` parameter. It's advantage is that it doesn't have any mental load compared to `SIZE`.